src/plugins/crypto/crypto.c

```diff
@@ -316,6 +422,62 @@ int CRYPTO_PLUGIN_FUNCTION (set) (Plugin * handle, KeySet * ks, Key * parentKey)
 }
 
 /**
+ * @brief Checks for the existense of the master password, that is used for encryption and decryption.
+ *
+ * If the master password can not be found it will be generated randomly.
+ * Then it will be encrypted and stored in conf.
+ *
+ * If the master password can be found, it will be decrypted temporarily in order to verify its correctness.
+ * conf will not be modified in this case.
+ *
+ * An error might occur during the password generation, encryption and decryption.
+ * The error will be appended to errorKey.
+ *
+ * @retval 0 no changes were made to the configuration
+ * @retval 1 the master password has been appended to the configuration
+ * @retval -1 an error occured. Check errorKey
+ */
+int CRYPTO_PLUGIN_FUNCTION (checkconf) (Key * errorKey, KeySet * conf)
+{
+	Key * k = ksLookupByName (conf, ELEKTRA_CRYPTO_PARAM_MASTER_PWD, 0);
+	if (k)
+	{
+		// call gpg module to verify that we own the required key
+		Key * msg = keyDup (k);
+		if (elektraCryptoGpgDecryptMasterPassword (conf, errorKey, msg) != 1)
+		{
+			ELEKTRA_SET_ERROR (ELEKTRA_ERROR_CRYPTO_CONFIG_FAULT, errorKey, "Master password decryption failed");
+			keyDel (msg);
+			return -1;
+		}
+		keyDel (msg);
+		return 0;
+	}
+	else
+	{
+		// generate random master password r
+		const kdb_unsigned_short_t passwordLen = elektraCryptoGetRandomPasswordLength (conf);
+		const char * r = elektraCryptoCreateRandomString (passwordLen);
+		if (!r)
+		{
+			ELEKTRA_SET_ERROR (ELEKTRA_ERROR_CRYPTO_RANDOM_GEN, errorKey, "Random password generation failed");
+			return -1;
+		}
+
+		// store password in configuration
+		k = keyNew ("user/" ELEKTRA_CRYPTO_PARAM_MASTER_PWD, KEY_END);
+		keySetString (k, r);
+		if (elektraCryptoGpgEncryptMasterPassword (conf, errorKey, k) != 1)
+		{
+			keyDel (k);
+			return -1;
+		}
+		ksAppendKey (conf, k);
+		return 1;
+	}
```

Comment:
Please also validate all other config options (see also note about not fail silently, its always best to already enforce correct config here)